api/outdated/outdated/tests/test_parser.py

```diff
@@ -116,3 +119,64 @@ def test_parser(db, tmp_repo_root, project, lockfile, content, expected):
 
     for result in results:
         assert str(result) in expected
+
+
+class Response:
+    def __init__(self, data):
+        self.data = data
+
+    def json(self):
+        return self.data
+
+
+@pytest.mark.django_db()
+@pytest.mark.parametrize(
+    "dependency_names,associations,call_count",
+    [
+        (
+            ["ember-data", "ember-cli", "ember-source", "ember-modifier"],
+            {
+                "ember-(data|source|cli)": "emberjs",
+            },
+            3,
+        ),
+        (
+            ["django", "django-restframework"],
+            {
+                "django": "django",
+            },
+            1,
+        ),
+        (
+            ["djangorestframework", "django-environ"],
+            {
+                "django": "django",
+            },
+            0,
+        ),
+    ],
+)
+def test_fetch_end_of_life(
+    settings, mocker, dependency_names, associations, call_count
+):
+    settings.ENDOFLIFE_DATE_ASSOCIATIONS = associations
+    mocker.patch.object(
+        LockfileParser, "_get_release_date", return_value=date(2024, 1, 1)
+    )
+    request_get_mock = mocker.patch.object(
+        requests,
+        "get",
+        return_value=Response(
+            {
+                "lts": True,
+                "releaseDate": "2023-04-03",
+                "support": "2023-12-04",
+                "eol": "2026-04-01",
+                "latest": "4.2.9",
+                "latestReleaseDate": "2024-01-02",
+            }
+        ),
+    )
```

Comment:
What's the reasoning behind not using `requests-mock` here? It's already in your dependencies. I see several (potential) downsides to this approach here compared to using `requests-mock`:

1. You're patching the global `requests.get()` (this could be resolved tho)
2. With `requests-mock`, there is no need to manually create a `Response`
3. It would be trivial to alter the response in order to provoke a `JSONDecodeError` in `LockfileParser._get_version():82` and get rid of the `pragma`

Reply:
1, 2 done here: https://github.com/adfinis/Outdated/pull/200/commits/e134387dd39fdee48928f8fb62e839463ca3c324